NitroxClient/GameLogic/Spawning/WorldEntities/OxygenPipeEntitySpawner.cs

```diff
@@ -0,0 +1,135 @@
+using System.Collections;
+using System.Collections.Generic;
+using NitroxClient.GameLogic.Spawning.Abstract;
+using NitroxClient.MonoBehaviours;
+using NitroxClient.Unity.Helper;
+using NitroxModel.DataStructures;
+using NitroxModel.DataStructures.GameLogic.Entities;
+using NitroxModel.DataStructures.Util;
+using NitroxModel_Subnautica.DataStructures;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic.Spawning.WorldEntities;
+
+public class OxygenPipeEntitySpawner : SyncEntitySpawner<OxygenPipeEntity>
+{
+    private readonly Entities entities;
+    private readonly WorldEntitySpawner worldEntitySpawner;
+
+    private readonly Dictionary<NitroxId, List<OxygenPipe>> childrenPipeEntitiesByParentId;
+
+    public OxygenPipeEntitySpawner(Entities entities, IEntitySpawner worldEntitySpawner)
+    {
+        this.entities = entities;
+        this.worldEntitySpawner = (WorldEntitySpawner)worldEntitySpawner;
+
+        childrenPipeEntitiesByParentId = new();
+    }
+
+    protected override IEnumerator SpawnAsync(OxygenPipeEntity entity, TaskResult<Optional<GameObject>> result)
+    {
+        if (!DefaultWorldEntitySpawner.TryGetCachedPrefab(out GameObject prefab, classId: entity.ClassId))
+        {
+            TaskResult<GameObject> prefabResult = new();
+            yield return DefaultWorldEntitySpawner.RequestPrefab(entity.ClassId, prefabResult);
+            if (!prefabResult.Get())
+            {
+                Log.Error($"Couldn't find a prefab for {nameof(OxygenPipeEntity)} of ClassId {entity.ClassId}");
+                yield break;
+            }
+            prefab = prefabResult.Get();
+        }
+
+        GameObject gameObject = GameObjectHelper.InstantiateInactiveWithId(prefab, entity.Id);
+        if (!IsSpawnedPrefabValid(entity, gameObject, out OxygenPipe oxygenPipe, out string errorLog))
+        {
+            Log.Error(errorLog);
```

Comment:
I don't see why the logging needs to be here. It complicates the code while it brings no value. If you need the StackTrace just log a exception instead of plain string.